compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3844,7 +3848,7 @@ private BType createRecordIntersection(IntersectionContext diagnosticContext,
 
     private BType getConstraint(BRecordType recordType) {
         if (recordType.sealed) {
-            return symTable.neverType;
+            return symTable.noType;
```

Comment:
@MaryamZi can you please verify this change?

Reply:
@lasinicl can you share an example of why we needed this change, please?

Reply:
```
type ClosedRecordWithIntField record {|
    int i;
|};
ClosedRecordWithIntField val5 = {i: 100};
boolean b10 = val5 is record {| int i; string s; |}; //incompatible types: 'ClosedRecordWithIntField' will not be matched to 'record {| int i; string s; |}'
```
When there's `never` here, the above compilation error was not given.
https://github.com/ballerina-platform/ballerina-lang/blob/master/tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/type-test-expr-negative.bal#L296

Reply:
This is happening due to the checks in org.wso2.ballerinalang.compiler.semantics.analyzer.Types#validateRecordField, right?

I think we should try and continue to use never here, and when validating the field explicitly check if the intersection type is never, and if not `Symbols.isOptional(origField.symbol)` return semantic error, or something similar to this.

Also, IMO, the behaviour should be the same for a closed record and an open record with a `never` rest field.

Both the type tests in the following example result in errors on alpha4, but only one results in an error on the current master.

```ballerina
type ClosedRecordWithIntField record {|
    int i;
|};

type RecordWithIntFieldAndNeverRestField record {|
    int i;
    never...;
|};

type Record record {| 
    int i; 
    string s; 
|};

public function main() {
    Record rec = {i: 1, s: ""};

    boolean b1 = rec is ClosedRecordWithIntField;
    boolean b2 = rec is RecordWithIntFieldAndNeverRestField; // no error
}
```

Reply:
Okay. Will fix this with 29814 PR :)